src/lib/datatypes/sol-buffer.c

```diff
@@ -868,3 +868,19 @@ sol_buffer_fini(struct sol_buffer *buf)
     buf->used = 0;
     buf->capacity = 0;
 }
+
+SOL_API struct sol_blob *
+sol_buffer_to_blob(struct sol_buffer *buf)
+{
+    struct sol_blob *blob;
+
+    SOL_NULL_CHECK(buf, NULL);
+
+    blob = sol_blob_new(SOL_BLOB_TYPE_DEFAULT, NULL, buf->data, buf->used);
```

Comment:
hmm, here i think we have two approaches.  Convert the buffer to blob (then you can steal the buffer memory) or create a blob from buffer `sol_blob_from_buffer(const struct sol_buffer *buf)`. The later you will copy the memory.